### PR TITLE
Remove old, unused Ledger

### PR DESCRIPTION
**Pull request by m4ksio**

### CODE REVIEW

File: ledger/ledger.go [hunk 1]
```diff
@@ -3,11 +3,14 @@ package ledger
 import (
 	"bytes"
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 
 	"github.com/onflow/flow-go/module"
 )
 
+const CacheSize = 1000
```
Comment (ramtinms) [inline ledger/ledger.go:12]: This shouldn't be here in my view, should be part of the complete ledger.

File: ledger/complete/wal/wal.go [hunk 2]
```diff
@@ -93,14 +93,15 @@ func (w *LedgerWAL) Replay(
...
+	checkpointFn func(forestSequencing *flattener.FlattenedForest) error,
 	updateFn func(update *ledger.TrieUpdate) error,
 	deleteFn func(rootHash ledger.RootHash) error,
 ) error {
 	from, to, err := w.wal.Segments()
 	if err != nil {
 		return err
 	}
-	return w.replay(from, to, nil, updateFn, deleteFn, false)
+	return w.replay(from, to, checkpointFn, updateFn, deleteFn, false)
```
Comment (ramtinms) [inline ledger/complete/wal/wal.go:104]: when replaying the transactions do we really want to use checkpointer ?

File: ledger/complete/mtrie/trie/trie.go [hunk 3]
```diff
@@ -384,6 +387,52 @@ func (mt *MTrie) Equals(o *MTrie) bool {
 	return o.PathLength() == mt.PathLength() && bytes.Equal(o.RootHash(), mt.RootHash())
 }
 
+// DumpAsJSON dumps the trie key value pairs to a file having each key value pair as a json row
+func (mt *MTrie) DumpAsJSON(path string) error {
```
Comment (ramtinms) [inline ledger/complete/mtrie/trie/trie.go:391]: this needs a test

File: ledger/complete/wal/wal.go (hunk 2 above)
Comment (m4ksio) [reply ledger/complete/wal/wal.go:104]: Yes, for possible `root.checkpoint`

File: ledger/ledger.go (hunk 1 above)
Comment (m4ksio) [reply ledger/ledger.go]: Yeah, you're right

File: ledger/complete/mtrie/trie/trie.go (hunk 3 above)
Comment (m4ksio) [reply ledger/complete/mtrie/trie/trie.go]: Added some base tests

**Review by ramtinms — APPROVED** [review]

**Pull request merged**